configure

```diff
@@ -4049,6 +4088,37 @@ fi
 define_in_config_if_yes "$_opengl" "USE_OPENGL"
 define_in_config_if_yes "$_opengles" "USE_GLES"
 
+if test "$_use_gles_custom" = yes ; then
+	_opengles=yes
+	add_line_to_config_mk "USE_GLES = 1"
+	add_line_to_config_h "#define USE_GLES"	
+	add_line_to_config_mk "USE_GLES_CUSTOM = 1"
+	add_line_to_config_h "#define USE_GLES_CUSTOM"
+	if test "$_use_gles_rpi" = yes ; then
+		echo "Activating custom-context GLES RPI"
+		# Yes, the Raspberry toolchain uses hardcoded paths. See Broadcom examples in /opt/vc/src.
+		CXXFLAGS="$CXXFLAGS -mfpu=vfp -mfloat-abi=hard -march=armv6zk -mtune=arm1176jzf-s -I/opt/vc/include -I/opt/vc/include/interface/vmcs_host/linux/ -I/opt/vc/include/interface/vcos/pthreads"
+		LIBS="$LIBS -L/opt/vc/lib -lGLESv1_CM -lEGL -lm -lbcm_host -lvcos -lvchiq_arm"
+		add_line_to_config_mk "USE_GLES_RPI = 1"
+		add_line_to_config_h "#define USE_GLES_RPI"
+	fi
+	
+	if test "$_use_gles_fbdev" = yes ; then
+		echo "Activating custom-context GLES FBDEV (Cubieboard/Cubie2/Cubietruck)"
+		CXXFLAGS="$CXXFLAGS -marm -mcpu=cortex-a7 -mfloat-abi=hard -mfpu=neon"
+		LIBS="$LIBS -L/opt/vc/lib -lGLESv1_CM -lEGL -lm"
+		add_line_to_config_mk "USE_GLES_FBDEV = 1"
+		add_line_to_config_h "#define USE_GLES_FBDEV"
+	fi
+	
+	if test "$_use_gles_kms" = yes ; then
+		echo "Activating custom-context GLES KMS (OpenGL(ES) without X)"
+		CXXFLAGS="$CXXFLAGS -I/usr/include/drm"
+		LIBS="$LIBS -L/opt/vc/lib -lGLESv1_CM -lEGL -lm -ldrm -lgbm"
+		add_line_to_config_mk "USE_GLES_KMS = 1"
+		add_line_to_config_h "#define USE_GLES_KMS"
+	fi
+fi
```

Comment:
Could you add a simple tiny test here to check validity of the command line parameters? I am really worrying about incompatibility with further versions of toolchains, so adding a simple compilation test could save time on development.